include/cuco/detail/open_addressing_ref_impl.cuh

```diff
@@ -63,12 +64,12 @@ class open_addressing_ref_impl {
                       ProbingScheme>,
     "ProbingScheme must inherit from cuco::detail::probing_scheme_base");
 
-  static_assert(is_window_extent_v<typename StorageRef::extent_type>,
-                "Extent is not a valid cuco::window_extent");
-  static_assert(ProbingScheme::cg_size == StorageRef::extent_type::cg_size,
-                "Extent has incompatible CG size");
-  static_assert(StorageRef::window_size == StorageRef::extent_type::window_size,
-                "Extent has incompatible window size");
+  // static_assert(is_window_extent_v<typename StorageRef::extent_type>,
+  //               "Extent is not a valid cuco::window_extent");
+  // static_assert(ProbingScheme::cg_size == StorageRef::extent_type::cg_size,
+  //               "Extent has incompatible CG size");
+  // static_assert(StorageRef::window_size == StorageRef::extent_type::window_size,
+  //               "Extent has incompatible window size");
```

Comment:
How could we solve the issue where the sum of `window_extent`s is not a `window_extent` itself?

Reply:
Losing these checks isn't ideal. We could create a new `window_extent` from the sum using `make_window_extent` and pass that to the ctor.

Reply:
> Losing these checks isn't ideal. 

Agreed. That's the complicated part.

In general, when users have a pointer and a size on hand. Creating a ref should be as simple as:
```cpp
auto ref = ref_type{&data, size};
```
That's what I'm trying to achieve for `storage_ref` in the [subset example](https://github.com/PointKernel/cuCollections/blob/085d1bb3fde848919913466f2f7855c1bec73f04/examples/static_set/device_subsets_example.cu#L100-L101):
```cuda
  auto set_ref = ref_type<cuco::experimental::find_tag>{
    cuco::empty_key<key_type>{-1}, {}, {}, storage_ref_type{sizes[idx], windows + offsets[idx]}};
```
Enabling those checks enforces users to invoke `make_window_extent` again over `sizes[i]` (Note `sizes[i]` is already the output of `make_window_extent`). Also, default template parameters are no longer valid either thus users need to specify them explicitly. The above code would turn into:
```cuda
  using extent_type =
    decltype(make_window_extent<cg_size, window_size>(std::declval<cuco::experimental::extent<size_t>>()));
  auto set_ref = ref_type<cuco::experimental::find_tag>{
    cuco::empty_key<key_type>{-1},
    {},
    {},
    aow_storage_ref<key_type,
                    window_size,
                    extent_type>{make_valid_extent<cg_size, window_size>(sizes[idx]), windows + offsets[idx]}};
```

This is way more complex than needed.

One solution I can think of is to set the proper data type for [sizes array](https://github.com/PointKernel/cuCollections/blob/085d1bb3fde848919913466f2f7855c1bec73f04/examples/static_set/device_subsets_example.cu#L134), so instead of:
```cuda
  auto valid_sizes = std::vector<std::size_t>(num);
```
Users should get the return type of `make_window_extent` first and then declare the array:
```cuda
  using extent_type =
    decltype(make_window_extent<cg_size, window_size>(std::declval<cuco::experimental::extent<size_t>>()));
  auto valid_sizes = std::vector<extent_type>(num);
```
One thing I don't like here is the obscure way that users have to follow to set up the proper extent type. Isn't all those fiddlings around `cg_size`, `window_size`, and `size_t`, etc too complicated for people who just want a size? By all means, this is a doable workaround but doesn't solve the core problem that we are prohibiting users to create a data ref with a pointer and a size.

Reply:
100% agree on the unnecessary complexity. It should be as simple as passing a pointer and a size (or a `cuda::std::span` once we have it).

Can we provide an additional ctor with signature `storage_ref_type(ptr, size)`, which then internally constructs a `window_extent` from the size?

We can discuss this in today's dev sync.

Reply:
> which then internally constructs a `window_extent` from the size?

We lose the motivation of having a `window_extent` strong type in that way.